components/brave_vpn/brave_vpn_service_desktop.cc

```diff
@@ -427,18 +466,25 @@ void BraveVpnServiceDesktop::LoadPurchasedState() {
   }
 #endif
 
-  const std::string credential =
-      prefs_->GetString(brave_rewards::prefs::kSkusVPNCredential);
-  if (skus_credential_ == credential)
-    return;
-
-  skus_credential_ = credential;
+  const bool has_credential =
+      prefs_->GetBoolean(skus::prefs::kSkusVPNHasCredential);
 
-  if (!skus_credential_.empty()) {
-    VLOG(2) << __func__ << " : "
-            << "Loaded cached skus credentials";
-    SetPurchasedState(PurchasedState::PURCHASED);
+  if (!has_credential) {
+    // TODO: we can show logic for person to login
+    // NOTE: we might save (to profile) if person EVER had a valid
+    // credential. If so, we may want to show an expired dialog
+    // instead of the "purchase" dialog.
+    skus_credential_ = "";
+    return;
   }
+
+  // if a credential is ready, we can present it
+  // name should be environment specific but prefix with "vpn."
+  // for now we can hardcode dev
+  skus_sdk_service_->PrepareCredentialsPresentation(
+      "vpn.brave.software", "*",
```

Comment:
TODO: needs to use proper environment here

Reply:
Fixed with https://github.com/brave/brave-core/pull/10847/commits/4fb49af75b38a477ef6eb691bc256f5b87ecfab0